core/modules/query/ValueFactor.h

```diff
@@ -106,14 +106,17 @@ class ValueFactor {
     std::string _tableStar; // Reused as const val (no tablestar)
 };
 
-class ValueFactor::render : public std::unary_function<ValueFactor, void> {
+
+class ValueFactor::render {
 public:
     render(QueryTemplate& qt) : _qt(qt) {}
-    void operator()(ValueFactor const& ve);
-    void operator()(ValueFactor const* vep) {
-        if(vep) (*this)(*vep); }
-    void operator()(std::shared_ptr<ValueFactor> const& vep) {
-        (*this)(vep.get()); }
+    void applyToQT(ValueFactor const& ve);
+    void applyToQT(ValueFactor const* vep) {
+        if(vep) applyToQT(*vep);
```

Comment:
I got lazy. This wasn't really a ticket about refactoring this stuff, but it was really not easy to read.